From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mina Her <minacle@live.com>
Date: Mon, 25 May 2026 12:25:25 +0900
Subject: [PATCH] Reorganise About page content

- Force RulesSection to remain open by default
- Move RulesSection below the header
- Shift server meta block to the bottom of the page

Rationale: Prioritising the visibility of community rules improves user engagement, while relocating administrative metadata creates a more logical information hierarchy for visitors.

diff --git a/app/javascript/mastodon/features/about/components/rules.tsx b/app/javascript/mastodon/features/about/components/rules.tsx
index 07ca103960..69952d5a6b 100644
--- a/app/javascript/mastodon/features/about/components/rules.tsx
+++ b/app/javascript/mastodon/features/about/components/rules.tsx
@@ -74,7 +74,7 @@ export const RulesSection: FC<RulesSectionProps> = ({ isLoading = false }) => {
 
   if (rules.length === 0) {
     return (
-      <Section title={intl.formatMessage(messages.rules)}>
+      <Section open title={intl.formatMessage(messages.rules)}>
         <p>
           <FormattedMessage
             id='about.not_available'
@@ -86,7 +86,7 @@ export const RulesSection: FC<RulesSectionProps> = ({ isLoading = false }) => {
   }
 
   return (
-    <Section title={intl.formatMessage(messages.rules)}>
+    <Section open title={intl.formatMessage(messages.rules)}>
       <ol className='rules-list'>
         {rules.map((rule) => (
           <li key={rule.id}>
diff --git a/app/javascript/mastodon/features/about/index.jsx b/app/javascript/mastodon/features/about/index.jsx
index 27f03b17cb..3b844fe306 100644
--- a/app/javascript/mastodon/features/about/index.jsx
+++ b/app/javascript/mastodon/features/about/index.jsx
@@ -86,21 +86,7 @@ class About extends PureComponent {
             <p><FormattedMessage id='about.powered_by' defaultMessage='Decentralized social media powered by {mastodon}' values={{ mastodon: <a href='https://joinmastodon.org' className='about__mail' target='_blank' rel='noopener'>Mastodon</a> }} /></p>
           </div>
 
-          <div className='about__meta'>
-            <div className='about__meta__column'>
-              <h4><FormattedMessage id='server_banner.administered_by' defaultMessage='Administered by:' /></h4>
-
-              <Account id={server.getIn(['contact', 'account', 'id'])} size={36} minimal />
-            </div>
-
-            <hr className='about__meta__divider' />
-
-            <div className='about__meta__column'>
-              <h4><FormattedMessage id='about.contact' defaultMessage='Contact:' /></h4>
-
-              {isLoading ? <Skeleton width='10ch' /> : <a className='about__mail' href={`mailto:${server.getIn(['contact', 'email'])}`}>{server.getIn(['contact', 'email'])}</a>}
-            </div>
-          </div>
+          <RulesSection />
 
           <Section open title={intl.formatMessage(messages.title)}>
             {extendedDescription.get('isLoading') ? (
@@ -123,8 +109,6 @@ class About extends PureComponent {
             ))}
           </Section>
 
-          <RulesSection />
-
           <Section title={intl.formatMessage(messages.blocks)} onOpen={this.handleDomainBlocksOpen}>
             {domainBlocks.get('isLoading') ? (
               <>
@@ -156,6 +140,22 @@ class About extends PureComponent {
             ))}
           </Section>
 
+          <div className='about__meta'>
+            <div className='about__meta__column'>
+              <h4><FormattedMessage id='server_banner.administered_by' defaultMessage='Administered by:' /></h4>
+
+              <Account id={server.getIn(['contact', 'account', 'id'])} size={36} minimal />
+            </div>
+
+            <hr className='about__meta__divider' />
+
+            <div className='about__meta__column'>
+              <h4><FormattedMessage id='about.contact' defaultMessage='Contact:' /></h4>
+
+              {isLoading ? <Skeleton width='10ch' /> : <a className='about__mail' href={`mailto:${server.getIn(['contact', 'email'])}`}>{server.getIn(['contact', 'email'])}</a>}
+            </div>
+          </div>
+
           <LinkFooter />
 
           <div className='about__footer'>
